tests/integration/test_smoke.py

```diff
@@ -285,15 +285,14 @@ def test_client_can_retry_request_properly():
 
 def _make_client_call_with_errors(client, operation_name, kwargs):
     operation = getattr(client, xform_name(operation_name))
-    original_send = adapters.HTTPAdapter.send
-    def mock_http_adapter_send(self, *args, **kwargs):
+    original_send = Endpoint._send
+    def mock_endpoint_send(self, *args, **kwargs):
         if not getattr(self, '_integ_test_error_raised', False):
             self._integ_test_error_raised = True
-            raise ConnectionError("Simulated ConnectionError raised.")
```

Comment:
While I get that we're updating this to use our new exceptions, I do think it's useful to keep a few integration tests around that use the old exceptions to ensure that behavior isn't changed.

I would also prefer we leave the existing integration tests alone and add entirely new ones that verify we can catch the new exceptions.

This would serve as a nice safety check to ensure the same integration tests in develop can run against this PR.

Reply:
Added an integration test for one of the exception types (Read timeout) to ensure in a real case we're catching the old exceptions. I also created some unit tests that verify a new raised exception will be caught by the old exception type.